packages/sdk-bundles-api/src/replay-orchestrator/sdk-to-bundle/execute-replay.ts

```diff
@@ -195,6 +195,11 @@ export interface ReplayExecutionOptions {
    * to ensure that we don't compare screenshots that were generated from replays with different project settings.
    */
   projectSettingsHash?: string;
+
+  /**
+   * Defalts to false. If true, Meticulous will allow websockets to pass through to the real server.
+   */
+  passThroughWebSockets?: boolean;
```

Comment:
👍

Reply:
Conceptually this should probably be part of network stubbing mode, and having it part of that may make things easier if and when we want to expose it in the UI. It'd also pave nicely for future in case users eventually want to disable/enable on a per URL basis, and we want to enable this configurability without breaking API. However we'd have to be careful not to change existing semantics and so be a breaking change. One option would be to [update NetworkStubbingMode](https://github.com/alwaysmeticulous/meticulous-sdk/blob/fb9259223cd79d29aaa8969ea76779c4c01fb605/packages/api/src/sdk-bundle-api/sdk-to-bundle/network-stubbing.ts#L1) to:

```ts
export type NetworkStubbingMode =
  | StubAllRequests
  | StubNonSSRRequests
  | CustomStubbing;

/**
 * The default mode. Stubs all requests, apart from ones for _next/static/ files.
 */
export interface StubAllRequests {
  type: "stub-all-requests";
}

/**
 * Stubs all requests apart from NextJS 13 /app dir requests to render server components, and requests for _next/static/ files.
 *
 * Used for NextJs 13 /app directory & server components.
 */
export interface StubNonSSRRequests {
  type: "stub-non-ssr-requests";
}

export interface CustomStubbing {
  type: "custom-stubbing";

  /*
   * Any request that matches any of these filters will be passed through to the real backend and not stubed.
   *
   * We use an allowlist / opt-in pattern to reduce risk (e.g. disabling stubbing for all requests, and then
   * running a replay against production rather than staging, without realizing that stubbing is disabled)
   */
  requestsToNotStub: RequestFilter[];
}

export interface RequestFilter {
  /**
   * If defined will filter to only match requests to a URL matching the
   * specified regex.
   *
   * Any JS regex that passes https://github.com/tjenkinson/redos-detector is supported.
   */
  urlRegex: string;

  /** Defaults to `{ fetch: true, xhr: true, webSockets: false }` */
  connectionTypes?: ConnectionTypesFilter; // <<< ADDED
}

export interface NoStubbing {
  type: "no-stubbing";
  
  /** Defaults to `{ fetch: true, xhr: true, webSockets: false }` */
  connectionTypes?: ConnectionTypesFilter; // <<< ADDED
}

interface ConnectionTypesFilter { // <<< ADDED
      fetch: boolean;
      xhr: boolean;
      webSockets: boolean;
}
```

However I think we probably wouldn't need to perfectly preserve current semantics, and most people who have completely disabled stubbing probably want to disable web socket stubbing too, and so we could have connectionTypes absent from NoStubbing and it defaults to completely disabling stubbing, and maybe even have the default value for connectionTypes on RequestFilter to be `{ fetch: true, xhr: true, webSockets: true }` which is a more expected default value.

Reply:
My instinct would be to err on the side of making the breaking change and doing:

```ts
export interface RequestFilter {
  ...

  /** Defaults to `{ fetch: true, xhr: true, webSockets: true }` */
  connectionTypes?: ConnectionTypesFilter; // <<< ADDED
}

export interface NoStubbing {
  type: "no-stubbing";
}
```

Reply:
Extending network stubbing mode and making the breaking change makes sense to me.